src/tests/Microsoft.PowerFx.Interpreter.Tests/ConfigTests.cs

```diff
@@ -424,6 +424,84 @@ private void RecalcEngine_Symbol_CultureInfo6_ThreadProc(CultureInfo culture)
         }
 
+        [Fact]
+        public void Powerfx_Collect_Without_Serielize()
+        {
+            var result = (BlankValue)Run_Collect_Worflow(false);
+            Assert.Equal("Blank()", result.ToString());
+        }
+
+        [Fact]
+        public void Powerfx_Collect_With_Serielize()
+        {
+            var result = (ErrorValue)Run_Collect_Worflow(true);
+            Assert.Equal("Runtime type mismatch", result.Errors[0].Message);
+        }
+
+        private FormulaValue Run_Collect_Worflow(bool serielize)
+        {
+            // Define initial FormulaValue
+            RecordType schemaType = RecordType.Empty();
+            schemaType = schemaType.Add("Col1", FormulaType.String);
+            schemaType = schemaType.Add("Col2", FormulaType.String);
+            var list = new List<RecordValue>();
+            var fields = new List<NamedValue>
+            {
+                new NamedValue("Col1", FormulaValue.New("Col1")),
+                new NamedValue("Col2", FormulaValue.New("Col2"))
+            };
+            list.Add(FormulaValue.NewRecordFromFields(schemaType, fields));
+            FormulaValue formulaValue = FormulaValue.NewTable(schemaType, list);
+
+            var recalEngine = new RecalcEngine();
+            var result = (StringValue)RunExpr("Last(collection).Col2", recalEngine, false, formulaValue, "collection");
+            Assert.Equal("Col2", result.Value);
+
+            // Getting the FormulaValue that is used to update variable for the next expression
+            // formulaValue is of type RecordsOnlyTableValuee
+            formulaValue = recalEngine.GetValue("collection");
+
+            recalEngine = new RecalcEngine();
+            if (serielize)
+            {
+                // Serielizing it as SetPowerCardMemory does and stores it in memoryContext
+                string serielizedFormulaValue = formulaValue.ToExpression();
+
+                // Deserielizing it
+                // After serielization and deserielization, the formulaValue has type of InMemoryTableValue
+                formulaValue = RunExpr(serielizedFormulaValue, recalEngine, true);
+            }
+
+            RunExpr("Collect(collection, {Col1:\"newCol1\"})", recalEngine, false, formulaValue, "collection");
+            return RunExpr("Last(collection).Col2", recalEngine, true);
+        }
+
+        private FormulaValue RunExpr(string expressionText, RecalcEngine engine, bool engineInit, FormulaValue formulaValue = null, string varName = "")
+        {
+            // Parser options for RecalEngine
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = true,
+                NumberIsFloat = true
+            };
+
+            // Define symbol table
+            var symbolTable = new SymbolTable();
+            symbolTable.EnableMutationFunctions();
+            var symbolValues = symbolTable.CreateValues();
+
+            if (!engineInit)
```

Comment:
>engineInit [](http://example.com/codeflow?start=17&length=10)

Please rename it to something that is clearer. 'engineInit' can mean "initializeEngine" or "engineAlreadyInitialized", which are exact opposites. From the code it seems to be the latter, but we shouldn't have to look at the code to understand what a parameter means.

Another option would be to call it "setVariableValue", which is even clearer.